app/components/project-settings/analysis-settings/vulnerability-list/index.ts

```diff
@@ -51,10 +82,111 @@ export default class ProjectSettingsAnalysisSettingsVulnerabilityListComponent e
     return this.vulnerabilityListResponse?.toArray() || [];
   }
 
+  get vulnerability() {
+    return this.selectedVulnerabilityPreference
+      ? this.store.peekRecord(
+          'vulnerability',
+          this.selectedVulnerabilityPreference.id
+        )
+      : null;
+  }
+
+  get overrideCriteriaOptions() {
+    return [
+      {
+        label: this.intl.t('allFutureAnalyses'),
+        value: ENUMS.ANALYSIS_OVERRIDE_CRITERIA.ALL_FUTURE_UPLOAD,
+      },
+    ];
+  }
+
+  get openOverrideEditDrawer() {
+    return Boolean(this.vulnerabilityDataModel);
+  }
+
+  get vulnerabilityDataModel() {
+    return this.selectedVulnerabilityPreference
+      ? {
+          model: this.selectedVulnerabilityPreference,
+          vulnerabilityName: this.vulnerability?.name || '',
+          computedRisk: this.selectedVulnerabilityPreference.risk,
+          isOverridden: this.selectedVulnerabilityPreference.riskOverridden,
+          overriddenRisk: this.selectedVulnerabilityPreference.risk,
+          overriddenRiskComment: this.selectedVulnerabilityPreference.comment,
+          overriddenBy: this.selectedVulnerabilityPreference.updatedBy,
+          overrideCriteria: ENUMS.ANALYSIS_OVERRIDE_CRITERIA.ALL_FUTURE_UPLOAD,
+          overrideCriteriaOptions: this.overrideCriteriaOptions,
+          overriddenOn: this.selectedVulnerabilityPreference.updatedDate,
+
+          ignoreVulnerabilityHelperText: this.intl.t(
+            'projectSettings.vulnerabilityPreference.ignoreVulnerabilityHelperText'
+          ),
+
+          overrideSuccessMessage: this.intl.t(
+            'projectSettings.vulnerabilityPreference.overrideSuccessMessage'
+          ),
+
+          resetConfirmComponent:
+            'project-settings/analysis-settings/vulnerability-list/reset-confirm',
+
+          resetOverrideHandler: this.resetPreference,
+          editSaveOverrideHandler: this.savePreference,
+        }
+      : null;
+  }
+
+  @action
+  handleOverrideEditDrawerClose() {
+    this.selectedVulnerabilityPreference = null;
+  }
+
+  @action
+  setVulnerabilityDataModel(
+    vulnerabilityPreference: VulnerabilityPreferenceModel
+  ) {
+    this.selectedVulnerabilityPreference = vulnerabilityPreference;
+  }
+
   @action loadVulnerabilityPreferences() {
     this.fetchVulnerabilityPreferences.perform();
   }
 
+  @action
+  async savePreference(risk: number, comment: string) {
+    const data = { risk, comment };
+
+    await this.saveVulnerabilityPreferences.perform(data);
+  }
+
+  @action
+  async resetPreference() {
+    const data = {
+      risk: null,
+      comment: null,
+    };
+
+    await this.saveVulnerabilityPreferences.perform(data);
+  }
+
+  saveVulnerabilityPreferences = task(
```

Comment:
Wrap function logic in try/catch block in case of errors.

Reply:
this is not wrapped in try/catch so that error can be caught at the usage level

Reply:
Got it.